packages/asset-swapper/src/utils/market_operation_utils/bancor_service.ts

```diff
@@ -0,0 +1,36 @@
+import { BigNumber } from '@0x/utils';
+import { SDK } from '@bancor/sdk';
+import { BlockchainType, Token } from '@bancor/sdk/dist/types';
+
+import { BancorQuoteData } from './types';
+
+function token(address: string, blockchainType: BlockchainType = BlockchainType.Ethereum): Token {
+    return {
+        blockchainType,
+        blockchainId: address,
+    };
+}
+
+export class BancorService {
+    // Bancor recommends setting this value to 2% under the expected return amount
+    public minReturnAmountBufferPercentage = 0.98;
+    private _sdk?: SDK;
+
+    constructor(public ethereumNodeEndpoint: string) {}
+
+    public async getSDKAsync(): Promise<SDK> {
+        if (!this._sdk) {
+            this._sdk = await SDK.create({ ethereumNodeEndpoint: this.ethereumNodeEndpoint });
+        }
+        return this._sdk;
+    }
+
+    public async getQuoteAsync(fromToken: string, toToken: string, amount: BigNumber): Promise<BancorQuoteData> {
+        const sdk = await this.getSDKAsync();
+        const { path, rate } = await sdk.pricing.getPathAndRate(token(fromToken), token(toToken), amount.toString());
+        return {
+            fillData: { path: path.map(p => p.blockchainId) }, // ethereum token addresses
```

Comment:
this may also give different paths given different amounts?
E.g $100 gives A->B, $500 gives A->B->C. The sampler expects these to all be the same "route/source" so can encounter errors trying to fill for $600 if A->B or A->B->C do not have sufficient liquidity.

We should see if there are SDK functions that allow you to efficiently query with a hint of a path, or like Balancer, select a few candidates and query those.

Reply:
I wonder if we can just use the fillData from the last fill node.

Reply:
we can query with a full path or with source + target token, but I don't think we can provide a partial path. 

I'm making changes so `fillData.path` is unique to the amount quoted, so that should resolve the fill data problem. But perhaps still inefficient with the number of RPC calls per quote (1 per quote amount).

Reply:
did a deeeeep dive with @dorothy-zbornak in current implementation of path optimisation. as suggested, we can use the `fillData` from the last node.